TestFoundation/TestURLRequest.swift

```diff
@@ -194,6 +195,15 @@ class TestURLRequest : XCTestCase {
         XCTAssertNil(originalRequest.allHTTPHeaderFields)
     }
 
+    func test_hash() {
```

Comment:
Can we test also altering all the entries that contribute to the hash?

Reply:
`NSObject.hash` doesn't allow for custom seeding, so the possibility of collisions makes this unfeasible for properties that have less than a handful possible values -- but we can check most of them!

I think I'll consolidate these hashing PRs to allow for the introduction of some helper functions.